File: src/ruby/min_max_recursivo.rb
# Mínimo e Máximo recursivo em Ruby
# Carlos Alves
# https://github.com/EuCarlos

def recursive_min_and_max(arr, min, max, index)
    min = arr[index] if arr[index] < min
    max = arr[index] if arr[index] > max

    if index < arr.length - 1
        recursive_min_and_max(arr, min, max, index + 1)
    else
        puts "Mínimo: #{min}"
        puts "Máximo: #{max}"
    end
end

list = [37, 53, 1, 43, 11, 49, 32, 55, 40, 47, 25, 4]
recursive_min_and_max(list, list[0], list[0], 0)

File: src/java/MinMaxRecursivo.java
public class MinMaxRecursivo {

    public static void main(String[] args){
        int[] arr = {9,8,13,33,18,20,88,14,40,77};
        MinMax(arr, arr[0],arr[0],0);
    }

    public static void MinMax(int vect[], int minimo, int maximo, int incide){
        if(vect[incide] < minimo ){
            minimo = vect[incide];
        }
        if(vect[incide] > maximo){
            maximo = vect[incide];
        }
        if(incide < vect.length-1 ){
            MinMax(vect, minimo,maximo, incide+1);
        }
        else {
            System.out.println("Mínimo :" + minimo + "\nMáximo: "+maximo);
        }
    }
}
